app/Helper/ConfigurationParser.php

```diff
@@ -4,35 +4,160 @@
+
+    /** @var array $configuration */
+    protected $configuration;
+    /** @var ComposerConfiguration|null $composerConfiguration */
+    protected $composerConfiguration;
+    /** @var PackageConfigurationAbstract $packageConfiguration */
+    protected $packageConfiguration;
+
 
     /**
+     * ConfigurationParser constructor.
+     *
      * @param $fullyQualifiedFileName
+     *
+     * @throws InvalidFileException
+     * @throws MissingConfigurationException
+     * @throws MissingFileException
+     * @throws MissingParameterException
+     */
+    public function __construct($fullyQualifiedFileName)
+    {
+        $this->configuration = $this->parseJsonFile($fullyQualifiedFileName);
+        $this->setPackageConfiguration();
+        $this->setComposerConfiguration();
+    }
+
+
+    /**
+     * Parse the json file with the given fully qualified filename
+     *
+     * @param $fullyQualifiedFileName
+     *
      * @return mixed
-     * @throws Exception
+     *
+     * @throws InvalidFileException
+     * @throws MissingFileException
      */
-    public static function parseJsonFile($fullyQualifiedFileName)
+    protected function parseJsonFile($fullyQualifiedFileName)
     {
         $fileContentRaw = file_get_contents($fullyQualifiedFileName);
         if ($fileContentRaw === false) {
-            throw new Exception('Could not load configuration file!');
+            throw new MissingFileException('Could not load configuration file!');
+        }
+
+        $fileContent = json_decode($fileContentRaw, true);
+
+        if (json_last_error() != JSON_ERROR_NONE) {
+            throw new InvalidFileException('Configuration file is not valid json!');
+        }
+
+        return $fileContent;
+    }
+
+    /**
+     * Fetch the given key from the loaded configuration
+     *
+     * @param string|int $key
+     *
+     * @return array|string|null
+     *
+     * @throws MissingParameterException
+     */
+    protected function fetchConfigurationParameter($key)
+    {
+        return ArrayHelper::valueByKey($this->configuration, $key);
+    }
+
+    /**
+     * Set the package configuration
+     *
+     * @throws MissingConfigurationException
+     * @throws MissingParameterException
+     */
+    protected function setPackageConfiguration()
+    {
+        $packageConfiguration = $this->fetchConfigurationParameter(self::CONFIGURATION_KEY_PACKAGE);
+        if ($packageConfiguration === null) {
+            throw new MissingConfigurationException('No package configuration provided at .github/artifact-configuration.json');
+        }
+
+        $this->packageConfiguration = (new PackageConfigurationBuilder())->build($packageConfiguration);
+
+    }
+
+    /**
+     * Set the composer configuration
+     *
+     * @throws MissingParameterException
+     */
+    protected function setComposerConfiguration()
+    {
+        $composerConfiguration = $this->fetchConfigurationParameter(self::CONFIGURATION_KEY_COMPOSER);
+
```

Comment:
if $this->composerConfiguration is null:
    $this->composerConfiguration = new InitClass?